web/packages/teleport/src/Roles/RoleEditor/StandardEditor/Resources.tsx

```diff
@@ -406,15 +467,29 @@ function KubernetesResourceView({
           />
         </ButtonIcon>
       </Flex>
-      <FieldSelect
-        label="Kind"
-        isDisabled={isProcessing}
-        options={kubernetesResourceKindOptions}
-        value={kind}
-        rule={precomputed(validation.kind)}
-        onChange={k => onChange?.({ ...value, kind: k })}
-        menuPosition="fixed"
+      <KubernetesResourceKindView
+        value={value}
+        validation={validation.kind}
+        isProcessing={isProcessing}
+        onChange={k => onChange?.({ ...value, ...k })}
+        roleVersion={value.roleVersion}
       />
+      <Box hidden={(!group || group === '*') && !supportsCRDs}>
```

Comment:
A hidden box that contains a field still uses that field's validation rule. I know that we are using validation in a different way in this form, but let's keep away from potentially error-prone code. Can we make it conditionally rendered?